Source/GoBot/ViewDatabase.swift

```diff
@@ -2013,5 +2036,41 @@ class ViewDatabase {
             ))
         }
     }
-
 } // end class
+
+fileprivate extension MessageIdentifier {
+    var sha256hash: Data {
+        if #available(iOS 13.0, *) {
+            let input = self.data(using: .utf8)!
+            let hashed = SHA256.hash(data: input)
+            // using description is silly but i couldnt figure out https://developer.apple.com/documentation/cryptokit/sha256digest Accessing Underlying Storage
+            let descr = hashed.description
+            let prefix = "SHA256 digest: "
+            guard descr.hasPrefix(prefix) else { fatalError("oh gawd") }
+            return Data(hexString: String(descr.dropFirst(prefix.count)))!
+        } else {
+            // https://augmentedcode.io/2018/04/29/hashing-data-using-commoncrypto/ ?
+            fatalError("TODO: get CommonCrypto method to work or find another swift 5 method")
```

Comment:
@martindsq i tried the older API for an hour and gave up.. If you have a pod up your sleve to do this, I'm all yours. 

OTOH Dropping support for iOS12 might also make sense at some point...?